sktime/networks/cnn.py

```diff
@@ -44,10 +51,12 @@ def __init__(
         avg_pool_size=3,
         n_conv_layers=2,
         activation="sigmoid",
+        padding="valid",
```

Comment:
An interesting point regarding the implementation is that the authors of this paper have not mentioned anything about padding ([link](https://ieeexplore.ieee.org/stamp/stamp.jsp?arnumber=7870510) to the paper, Ctrl + F to search for pad, padding etc.). "valid" seems to be the choice of the authors of the code.

The hard-coded "same" might be to pass the CI tests since this architecture might result in negative dims for smaller data sizes. We could face the same problem which makes me question what should our default value be.

"valid" seems like the more apt choice (this is also the default value of the underlying Conv1D) but this would mean passing "same" into test configs (class method called `get_test_params()`) of `CNNClassifier` and `CNNRegressor`.

I'd love your and @fkiraly's opinion on it.

Reply:
Hi @achieveordie,

I wanted to share my thoughts on the default value for padding. Firstly, thank you for providing the link to the paper. After considering the options, I believe using `valid` padding as the default choice may not be ideal. This is because, as you mentioned, it can lead to negative dimensions and potentially leaving out important edge information in certain cases.

On the other hand, `same` padding appears to be a more rational choice. Although it may increase resources consumption and complexity, it ensures that the dimensions remain consistent, avoiding the issue of negative dimensions and preserving information at the edges.

However, I must admit that my expertise in CNNs for time series prediction is limited. I would greatly appreciate hearing your and @fkiraly thoughts on this matter.